### PR TITLE
D8CORE-2521: Config updates for embeddables.

### PR DESCRIPTION
**Pull request by imonroe**
# Ready for review

# Summary
- These are the config changes needed for the embeddable media type upgrade in `stanford_media`

# Needed By (Date)
- Midnight tonight

# Urgency
- it is critical for the embedded media functionality to work.

# Steps to Test

1. Do this
1. Then this
2. Then this

# Affected Projects or Products
- `stanford_media`

# Associated Issues and/or People
- D8CORE-2521
- https://github.com/SU-SWS/stanford_media/pull/61

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: config/sync/user.role.anonymous.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ permissions:
   - 'access content'
   - 'access patterns page'
   - 'search content'
+  - 'view field_media_embeddable_code'
   - 'view media'
+  - 'view own field_media_embeddable_code'
```
Comment (sherakama) [inline config/sync/user.role.anonymous.yml:17]: Don't need this perm as they won't be able to create media.

Comment (sherakama) [reply config/sync/user.role.anonymous.yml:17]: meh really.

File: config/sync/user.role.site_editor.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ permissions:
   - 'administer nodes'
   - 'administer stanford-event-types menu items'
   - 'administer stanford-person-type menu items'
+  - 'create embeddable media'
```
Comment (sherakama) [inline config/sync/user.role.site_editor.yml:26]: Shouldn't the contributor role also be able to create embeddable media?

File: config/sync/media.type.embeddable.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    - Instagram
+    - Issuu
+    - Livestream
+    - MathEmbed
+    - Simplecast
+    - SlideShare
+    - SoundCloud
+    - Spotify
+    - 'Stanford Digital Repository'
+    - Twitter
```
Comment (sherakama) [inline config/sync/media.type.embeddable.yml:32]: Will need to validate that these all work.

**Comment by sherakama** [comment]
These will need a few codeception tests to validate the existence of the embeddable stuff and validate the appropriate permission grants for the different roles.

File: config/sync/core.entity_form_display.media.embeddable.default.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    - field.field.media.embeddable.field_media_embeddable_oembed
+    - media.type.embeddable
+  module:
+    - media
+    - path
+id: media.embeddable.default
+targetEntityType: media
+bundle: embeddable
+mode: default
+content:
```
Comment (pookmish) [inline config/sync/core.entity_form_display.media.embeddable.default.yml:16]: Hide all the fields that aren't necessary.

File: config/sync/core.entity_form_display.media.embeddable.media_library.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    - field.field.media.embeddable.field_media_embeddable_code
+    - field.field.media.embeddable.field_media_embeddable_oembed
+    - media.type.embeddable
+  module:
+    - path
+id: media.embeddable.media_library
+targetEntityType: media
+bundle: embeddable
+mode: media_library
+content:
```
Comment (pookmish) [inline config/sync/core.entity_form_display.media.embeddable.media_library.yml:16]: Hide all the fields that aren't necessary.

File: config/sync/core.entity_view_display.media.embeddable.media_library.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - image.style.medium
+    - media.type.embeddable
+  module:
+    - image
+    - media
+id: media.embeddable.media_library
+targetEntityType: media
+bundle: embeddable
+mode: media_library
+content:
```
Comment (pookmish) [inline config/sync/core.entity_view_display.media.embeddable.media_library.yml:18]: Hide everything except the thumbnail and the name.

File: config/sync/media.type.embeddable.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    - stanford_media
+id: embeddable
+label: Embeddable
+description: 'Stanford Embeds'
+source: embeddable
+queue_thumbnail_downloads: false
+new_revision: false
+source_configuration:
+  thumbnails_directory: 'public://oembed_thumbnails'
+  providers:
```
Comment (pookmish) [inline config/sync/media.type.embeddable.yml:15]: Why don't these pull from the eoembed providers module config.?

File: config/sync/field.storage.media.field_media_embeddable_oembed.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+uuid: 36c142ba-b550-414d-824c-c5ed220eb0f9
+langcode: en
+status: true
+dependencies:
+  module:
+    - field_permissions
+    - media
+third_party_settings:
+  field_permissions:
+    permission_type: public
```
Comment (pookmish) [inline config/sync/field.storage.media.field_media_embeddable_oembed.yml:10]: Remove the field permissions if they are public. I know it does it automatically but its cleaner if we can.

**Review by pookmish — CHANGES_REQUESTED** [review]

File: config/sync/media.type.embeddable.yml (hunk 7 above)
Comment (imonroe) [reply config/sync/media.type.embeddable.yml:15]: So, for oEmbeds, you have to hard-code a list of providers that the plugin may support (https://github.com/SU-SWS/stanford_media/pull/61/files/53523547773564b25447fdb24c0f6a168f3f8bc7#r483181964) , and then in the config for the media type, you have the option enable or disable each of those supported providers individually.  These values determine which of our providers we will turn on.

File: config/sync/user.role.site_editor.yml (hunk 2 above)
Comment (imonroe) [reply config/sync/user.role.site_editor.yml:26]: when we had the conversation about permissions, I recall we decided that was not going to be the case, but if that's different at this point, it's no big deal to change it.

Comment (sherakama) [reply config/sync/user.role.site_editor.yml:26]: That is fine with me. Keep it this way.

File: config/sync/core.entity_form_display.media.embeddable.default.yml (hunk 4 above)
Comment (imonroe) [reply config/sync/core.entity_form_display.media.embeddable.default.yml:15]: fixed!

File: config/sync/core.entity_form_display.media.embeddable.media_library.yml (hunk 5 above)
Comment (imonroe) [reply config/sync/core.entity_form_display.media.embeddable.media_library.yml:14]: fixed!

File: config/sync/core.entity_view_display.media.embeddable.media_library.yml (hunk 6 above)
Comment (imonroe) [reply config/sync/core.entity_view_display.media.embeddable.media_library.yml:17]: fixed!

File: config/sync/field.storage.media.field_media_embeddable_oembed.yml (hunk 8 above)
Comment (imonroe) [reply config/sync/field.storage.media.field_media_embeddable_oembed.yml]: fixed!

File: stanford_profile.post_update.php [hunk 9]
```diff
@@ -46,3 +49,57 @@ function stanford_profile_post_update_8014() {
   ]);
 
 }
+
+/**
+ * Create blocks for sites with custom themes that were added to stanford_basic.
+ */
+function stanford_profile_post_update_8015() {
```
Comment (pookmish) [inline stanford_profile.post_update.php:56]: This was done in another PR wasn't it?

Comment (sherakama) [reply stanford_profile.post_update.php:56]: It went into this branch.

Comment (sherakama) [reply stanford_profile.post_update.php:56]: Suppose I could have pushed it into the main branch but I was building the deployments from this so it was easier

**Review by pookmish — APPROVED** [review]
